mne_icalabel/megnet/features.py

```diff
@@ -0,0 +1,291 @@
+
+    # Adjust coordinates with convex hull interpolation
+    hull = ConvexHull(channel_locations_2d)
+    border_indices = hull.vertices
+    Dborder = 1 / newR[border_indices]
+
+    funcTh = np.hstack(
+        [
+            TH[border_indices] - 2 * np.pi,
+            TH[border_indices],
+            TH[border_indices] + 2 * np.pi,
+        ]
+    )
+    funcD = np.hstack((Dborder, Dborder, Dborder))
+    interp_func = interpolate.interp1d(funcTh, funcD)
+    D = interp_func(TH)
+
+    adjusted_R = np.array([min(newR[i] * D[i], 1) for i in range(len(mags))])
+    Xnew, Ynew = pol2cart(adjusted_R, TH)
+    pos_new = np.vstack((Xnew, Ynew)).T
+
+    outlines = _make_head_outlines(np.array([0, 0, 0, 1]), pos_new, (0, 0))
+    return pos_new, outlines
+
+
+def _get_topomaps(ica: ICA, pos_new: NDArray, outlines: dict):
+    """Generate topomap images for each ICA component."""
+    topomaps = []
+    data_picks = mne.pick_types(ica.info, meg="mag")
+    components = ica.get_components()
+
+    for comp in range(ica.n_components_):
+        data = components[data_picks, comp]
+        fig = plt.figure(figsize=(1.3, 1.3), dpi=100, facecolor="black")
+        ax = fig.add_subplot(111)
+        mnefig, _ = mne.viz.plot_topomap(
+            data,
+            pos_new,
+            sensors=False,
+            outlines=outlines,
+            extrapolate="head",
+            sphere=[0, 0, 0, 1],
+            contours=0,
+            res=120,
+            axes=ax,
+            show=False,
+            cmap="bwr",
+        )
+        img_buf = io.BytesIO()
+        mnefig.figure.savefig(
+            img_buf, format="png", dpi=120, bbox_inches="tight", pad_inches=0
+        )
+        img_buf.seek(0)
+        rgba_image = Image.open(img_buf)
+        rgb_image = rgba_image.convert("RGB")
+        img_buf.close()
+        plt.close(fig)
+
+        topomaps.append(np.array(rgb_image))
+
+    return np.array(topomaps)
+
+
+def _line_noise_channel(
+    raw: BaseRaw,
+    picks: str,
+    fline: float = 50.0,
+    neighbor_width: float = 2.0,
+    threshold_factor: float = 3,
+    show: bool = False,
+):
+    """Detect line noise in MEG/EEG data.
+
+    Parameters
+    ----------
+    raw : mne.io.Raw
+        The raw MEG/EEG data.
+    picks : str or list, optional
+        Channels to include in the analysis.
+    fline : float, optional
+        The base frequency of the line noise to detect.
+    neighbor_width : float, optional
+        Width of the frequency neighborhood around each harmonic
+        for calculating background noise, in Hz. Default is 2.0 Hz.
+    threshold_factor : float, optional
+        Multiplicative factor for setting the detection threshold.
+        The threshold is set as the mean of the neighboring frequencies plus
+        `threshold_factor` times the standard deviation. Default is 3.
+    show : bool, optional
+        Whether to plot the PSD for channels affected by line noise.
+
+    Returns
+    -------
+    bool
```

Comment:
I made a mistake here, this function returns a `list` contains channels with line noise